components/auth/src/main/java/org/cloudfoundry/credhub/config/auth/OAuth2AuthenticationExceptionHandler.java

```diff
@@ -30,17 +31,29 @@ public void commence(HttpServletRequest request,
             throws IOException {
         btaep.commence(request, response, authException);
 
+        if (authException == null) {
+            return;
+        }
+
+        var jsonObject = new JsonObject();
+
         if (authException instanceof OAuth2AuthenticationException) {
-            OAuth2Error error = ((OAuth2AuthenticationException) authException)
-                    .getError();
-            response.setContentType("application/json");
-            var jsonObject = new JsonObject();
+
+            OAuth2Error error = ((OAuth2AuthenticationException) authException).getError();
             jsonObject.addProperty("error", error.getErrorCode());
             jsonObject.addProperty("error_description", error.getDescription());
-            try (PrintWriter writer = response.getWriter()) {
-                writer.write(jsonObject.toString());
-                writer.flush();
-            }
+
+        } else {
```

Comment:
So what are the types of `authException`s that are not `new-oauth2-mtls-exception`?

Reply:
The one that the 2 mtls acceptance tests were producing is InsufficientAuthenticationException.

Reply:
Are we sure that we want to populate the response body not just for `InsufficientAuthenticationException` but for all `AuthenticationException`?

Reply:
Unsure, figured it was a more consistent response content behavior to reply with the Exception message for all.  I don't currently see a downside.